apps/src/applab/applab.js

```diff
@@ -2905,6 +2946,11 @@ Applab.handleReadRecords = function(successCallback, records) {
 };
 
 Applab.updateRecord = function (opts) {
+  apiValidateType(opts, 'updateRecord', 'table', opts.table, 'string');
+  apiValidateType(opts, 'updateRecord', 'record', opts.record, 'object');
+  apiValidateTypeAndRange(opts, 'updateRecord', 'record.id', opts.record.id, 'number', 1, Infinity);
```

Comment:
It seems unlikely that a student would end up with a float here. However, borrowing from your other suggestion, I would propose allowing numerical strings (e.g. '11') anywhere we require a number (see updates to this PR). Let me know what you think.